odoo/cli/db.py

```diff
@@ -0,0 +1,152 @@
+        rename.set_defaults(func=self.rename)
+        rename.add_argument(
+            '-f', '--force', action='store_const', default=False, const=True,
+            help="delete `target` database before renaming if it exists"
+        )
+        rename.add_argument('source')
+        rename.add_argument("target", help="database to rename `source` to, must not exist unless `-f` is specified, in which case it will be dropped first")
+
+        drop = subs.add_parser("drop", help="Delete a database including filestore")
+        drop.set_defaults(func=self.drop)
+        drop.add_argument("database", help="database to delete")
+
+        args = parser.parse_args(cmdargs)
+
+        config.parse_config([
+            val
+            for k, v in vars(args).items()
+            if v is not None
+            if k in ['config', 'data_dir', 'addons_path'] or k.startswith(('db_', 'pg_'))
+            for val in [
+                '--data-dir' if k == 'data_dir'\
+                    else '--addons-path' if k == 'addons_path'\
+                    else f'--{k}',
+                v,
+            ]
+        ])
+        # force db management active to bypass check when only a
+        # `check_db_management_enabled` version is available.
+        config['list_db'] = True
+        report_configuration()
+
+        args.func(args)
+
+    def load(self, args):
+        url = urllib.parse.urlparse(args.dump_file)
+        if url.scheme:
+            stderr.write(f"Fetching {args.dump_file}...")
+            r = requests.get(args.dump_file, timeout=10, stream=True)
+            if not r.ok:
+                exit(f"Unable to fetch {args.dump_file}: {r.reason}")
+
+            stderr.write(" done\n")
+            dump_file = r.raw
+        else:
+            stderr.write(f"Restoring {args.dump_file}...\n")
+            dump_file = args.dump_file
+
+        restore_db(
+            db=args.database or Path(args.dump_file).stem,
+            dump_file=dump_file,
+            copy=True
+        )
+
+    def dump(self, args):
+        if args.dump_path == '-':
+            dump_db(args.database, stdout.buffer)
+        else:
+            with open(args.dump_path, 'wb') as f:
+                dump_db(args.database, f)
+
+    def duplicate(self, args):
+        self._check_target(args.target, delete_if_exists=args.force)
+        exp_duplicate_database(args.source, args.target)
+
+    def rename(self, args):
+        self._check_target(args.target, delete_if_exists=args.force)
+        exp_rename(args.source, args.target)
+
+    def drop(self, args):
+        if not exp_drop(args.database):
+            exit(f"Database {args.database} does not exist.")
+
+    def _check_target(self, target, *, delete_if_exists):
+        if exp_db_exist(target):
+            if delete_if_exists:
+                exp_drop(target)
+            else:
+                exit(f"Target database {target} exists, aborting.")
```

Comment:
We could simplify this by removing the delete_if_exists parameters using args.force

This will also allow to says "use --force to drop the existing database" in the message in order to help the user.

Reply:
> We could simplify this by removing the delete_if_exists parameters using args.force

That would require passing the entire `args` to the helper and I don't want to do that because it makes the interface shit.

Reply:
Making the args global (or an attribute) would solve this issue by removing args parameter everywhere.

Reply:
Making the interface even worse is not what I had in mind.